### PR TITLE
[ Term Entry ] .defineProperty() for Javascript

### PR DESCRIPTION
**Pull request by alimalim77**
### Description

Added the definition, syntax and example for .defineProperty().

### Type of Change

- Closes Issue: https://github.com/Codecademy/docs/issues/4026
- Adding a new entry
-  Created an example and added the same

### Checklist

- [✓ ] All writings are my own.
- [ ✓] My entry follows the Codecademy Docs style guide.
- [ ✓] My changes generate no new warnings.
- [ ✓] I have performed a self-review of my own writing and code.
- [ ✓] I have checked my entry and corrected any misspellings.
- [ ✓] I have made corresponding changes to the documentation if needed.
- [ ✓] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ✓] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ✓] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+Title: '.defineProperty()'
+Description: 'defines a property of an object either by creating or modifying.'
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:3]:
```suggestion
Description: 'Defines a property of an object either by creating or modifying.'
```

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  - 'Objects'
+  - 'Methods'
+  - 'Values'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/full-stack-engineer-career-path'
+---
+
+
+The `.defineProperty()` method defines a property on an object, either creating a new property or modifying an existing one. The method returns the modified object with the specified property.
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:17]:
```suggestion
The **`.defineProperty()`** method defines a property on an object, either creating a new property or modifying an existing one. The method returns the modified object with the specified property.
```

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Syntax
+
+```pseudo
+Object.defineProperty(obj, prop, descriptor)
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:22]: give bullet points below this explaining what are prop, obj and descriptor

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+The syntax above is used to define or modify a property on the object obj. The method returns the updated object containing the specified property.
+
+## Example
+
+In the example below, a `name` property is defined using the .defineProperty() method. Then, to check the value of property, by printing it to the console:
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:29]:
```suggestion
In the example below, a `name` property is defined using the `.defineProperty()` method. Then, to check the value of the property, by printing it to the console:
```

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+// Define a property 'name' on the 'person' object with a descriptor
+Object.defineProperty(person, 'name', {
+  value: 'John',
+  writable: true,
+  enumerable: true,
+  configurable: true,
+});
+
+console.log("Name:", person.name); // Output: John
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:42]:
```suggestion
console.log("Name:", person.name);
```

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+The syntax above is used to define or modify a property on the object obj. The method returns the updated object containing the specified property.
+
+## Example
+
+In the example below, a `name` property is defined using the .defineProperty() method. Then, to check the value of property, by printing it to the console:
+
+```js
+const person = {};
+
+// Define a property 'name' on the 'person' object with a descriptor
```
Comment (dakshdeepHERE) [inline content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:34]:
```suggestion
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have some comments for you to work on @alimalim77

File: content/javascript/concepts/objects/terms/definedProperty/defineProperty.md (hunk 1 above)
Comment (alimalim77) [reply content/javascript/concepts/objects/terms/definedProperty/defineProperty.md:3]: Resolved.

**Comment by alimalim77** [comment]
I have made the suggested changes @dakshdeepHERE . Please let me know if there is room for any more changes. Thanks!

**Review by dakshdeepHERE — APPROVED** [review]
Thank you for the commits @alimalim77. This looks good for second review.

**Review by avdhoottt — APPROVED** [review]
Looks good to get merged.

**Pull request merged**